src/interface/block_chain.cpp

```diff
@@ -529,33 +529,27 @@ void block_chain::fetch_block(const hash_digest& hash,
     handler(error::success, message, height);
 }
 
-void block_chain::fetch_block_txs_size(const hash_digest& hash,
-    block_txs_size_fetch_handler handler) const
+void block_chain::fetch_header_txs_size(const hash_digest& hash,
+    header_txs_size_fetch_handler handler) const
 {
 
     if (stopped())
     {
-        handler(error::service_stopped, nullptr, 0,std::vector<hash_digest>(),0);
+        handler(error::service_stopped, nullptr, 0, std::make_shared<hash_list>(std::vector<hash_digest>()),0);
         return;
     }
 
     const auto block_result = database_.blocks().get(hash);
 
     if (!block_result)
     {
-        handler(error::not_found, nullptr, 0,std::vector<hash_digest>(),0);
+        handler(error::not_found, nullptr, 0, std::make_shared<hash_list>(std::vector<hash_digest>()),0);
         return;
     }
 
     const auto height = block_result.height();
-    const auto tx_hashes = block_result.transaction_hashes();
-    const auto& tx_store = database_.transactions();
-    transaction::list txs;
-    txs.reserve(tx_hashes.size());
-    DEBUG_ONLY(size_t position = 0;)
-
-    const auto message = std::make_shared<const block>(block_result.header(),
-        std::move(txs));
+    const auto message = std::make_shared<const header>(block_result.header());
```

Comment:
Please add a `TODO` for remembering encapsulate the header and the tx list in a single object. It is more efficient, only one make_shared()...